sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -692,6 +693,45 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         tasks.RemoveAll(t => t.IsCompleted);
                     }
                 }
+
+                // If the main loop is aborting due to the connection being canceled, then
+                // force the processor to stop.
+                if ((!cancellationToken.IsCancellationRequested) && (Connection.IsClosed))
+                {
+                    Logger.ProcessorClientClosedException(Identifier);
+
+                    // Because this is a highly unusual situation
+                    // and goes against the goal of resiliency for the processor, surface a
+                    // unique exception with explicit message, passing the standard
+                    // ObjectDisposedException as the inner exception.
+                    var fatalException = new ServiceBusException(
```

Comment:
why not just throw ObjectDisposedException directly rather than wrapping it in SBE?

Reply:
I was trying to differentiate scenarios as we typically recover from `ObjectDisposedException` by starting a new task and continuing on.  I supposed we could go that way and use the message alone as the signal;  I can't say that there's really any better significance there to the `ServiceBusException`.

Reply:
Yeah I think throwing ObjectDisposedException is more appropriate here since we don't actually talk to the service.

Reply:
Fair enough;  I don't have a strong opinion either way, so `ObjectDisposeException` it is!